### DISCUSSION TITLE
Add extensions function to createHandler

### DISCUSSION THREAD

**Discussion by bryanpyle** [graphql/graphql-http#109] [Q&A] [answered]
### Story

As a server, I want to add an `extensions` callback function similar to what's available in the [express-graphql package options](https://www.npmjs.com/package/express-graphql#options)
so that I can add additional key-value metadata to each response. This is useful for capturing runtime and metrics for a given request. 

### Acceptance criteria

- The `graphql-http` `createHandler` function should have an optional `extensions` function for adding additional metadata to the GraphQL response as a key-value object. The result will be added to the "extensions" field in the resulting JSON. This is often a useful place to add development time metadata such as the runtime of a query or the amount of resources consumed. This may be an async function. The function is given one object as an argument: { document, variables, operationName, result, context }.

**Comment by enisdenjo** [comment, marked as answer]
This is already possible with the [`onOperation` handler hook](https://github.com/graphql/graphql-http/blob/main/docs/interfaces/handler.HandlerOptions.md#onoperation). Roughly:

```js
import { createHandler } from 'graphql-http';
import { schema } from './my-graphql-schema';

const handler = createHandler({
  schema,
  onOperation(_req, _args, result) {
    return {
      ...result,
      extensions: {
        myCustom: 'metadata',
      },
    };
  },
});
```

**Reply by bryanpyle** [reply]
Awesome! I'll give this a shot. Thank you!

**Reply by juampynr** [reply]
Worked for me. Thanks!
